### PR TITLE
TCVP-2036 added /email/reset endpoint to oracle-data-api

### PR DESCRIPTION
**Pull request by KDNeufeld**
# Description

This PR includes the following proposed change(s):

TCVP-2036 
Added new endpoint to reset a Disputant's email address
- added junit test to confirm code coverage and functionality

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/DisputeController.java [hunk 1]
```diff
@@ -263,6 +263,26 @@ public ResponseEntity<Void> verifyDisputeEmail(@PathVariable(name="id") @Paramet
 		return ResponseEntity.ok().build();
 	}
 
+	@Operation(summary = "Updates the email address of a particular Dispute.")
+	@ApiResponses({
+		@ApiResponse(responseCode = "200", description = "Ok. Email unverified."),
+		@ApiResponse(responseCode = "404", description = "Dispute with specified id not found"),
```
Comment (karahanb) [inline src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/DisputeController.java:269]: I think we should also add 400 bad request as a potential API response.

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/DisputeController.java [hunk 2]
```diff
@@ -263,6 +263,26 @@ public ResponseEntity<Void> verifyDisputeEmail(@PathVariable(name="id") @Paramet
...
+		@ApiResponse(responseCode = "500", description = "Internal server error occured.")
+	})
+	@PutMapping("/dispute/{id}/email/reset")
+	public ResponseEntity<Void> resetDisputeEmail(
+			@PathVariable(name="id")
+			@Parameter(description = "The id of the Dispute to update.")
+			Long id,
+			@RequestParam(name="email", required=true)
```
Comment (karahanb) [inline src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/DisputeController.java:277]: I remember from our discussion before that disputant will be able to provide blank email to remove previously provided email address and opt out using email. So in this case should we make this parameter required=false?

**Review by lfdgov — APPROVED** [review]

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/DisputeController.java (hunk 1 above)
Comment (KDNeufeld) [reply src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/DisputeController.java:269]: Yes, that's a good call. Missed that, thanks. Will add.

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/DisputeController.java (hunk 2 above)
Comment (KDNeufeld) [reply src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/DisputeController.java:277]: Ohhhh, right. Yes will remove that required=false.

**Review by karahanb — APPROVED** [review]
Great work! Thanks Kevin.

**Pull request merged**